yesod-test/Yesod/Test.hs

```diff
@@ -598,6 +602,29 @@ htmlCount query count = do
   liftIO $ flip HUnit.assertBool (matches == count)
     ("Expected "++(show count)++" elements to match "++T.unpack query++", found "++(show matches))
 
+-- | Parses the response body from JSON into a Haskell value, throwing an error if parsing fails.
+--
+-- This function also checks that the @Content-Type@ of the response is @application/json@.
+--
+-- ==== __Examples__
+--
+-- > get CommentR
+-- > (comment :: Comment) <- requireJSONResponse
+--
+-- @since 1.6.9
+requireJSONResponse :: (HasCallStack, FromJSON a) => YesodExample site a
+requireJSONResponse = do
+  withResponse $ \(SResponse _status headers body) -> do
+    let mContentType = lookup hContentType headers
+        isJSONContentType = maybe False (\contentType -> BS8.takeWhile (/= ';') contentType == "application/json") mContentType
```

Comment:
Copied from `acceptsJson`, mostly

Reply:
Would it work to use [`acceptsJson`](https://www.stackage.org/haddock/lts-14.16/yesod-core-1.6.16.1/Yesod-Core-Json.html#v:acceptsJson) here? That would be preferable, since the set of acceptable JSON mime types may expand in the future.

Reply:
`acceptsJson` is `MonadHandler`, so we couldn't just drop it in here I don't think, but I could create a new function exposed from yesod-core that takes a list of headers (`[(CI ByteString, ByteString)]`, aka [`RequestHeaders`](https://www.stackage.org/haddock/lts-14.16/http-types-0.12.3/Network-HTTP-Types-Header.html#t:RequestHeaders)) and returned if one of the was a JSON content type?

Reply:
That sounds good

Reply:
Realized that the `Accept` header is slightly different, since it has a list of preferred MIME types. I added a `contentTypeIsJson` function for general use on Content-Type.